cdap-security/src/main/java/io/cdap/cdap/security/authorization/AccessControllerWrapper.java

```diff
@@ -0,0 +1,279 @@
+import java.util.Collections;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+/**
+ * Wraps an {@link AccessController} and creates an {@link AccessControllerSpi} out of it.
+ */
+public class AccessControllerWrapper implements AccessControllerSpi {
+
+  private final AccessController accessControllerV1;
+
+  public AccessControllerWrapper(AccessController accessControllerV1) {
+    this.accessControllerV1 = accessControllerV1;
+  }
+
+  @Override
+  public void initialize(AuthorizationContext context) {
+    try {
+      accessControllerV1.initialize(context);
+    } catch (Exception e) {
+      throw Throwables.propagate(e);
+    }
+  }
+
+  @Override
+  public AuthorizationResponse createRole(Principal caller, Role role) throws AccessException {
+    try {
+      accessControllerV1.createRole(role);
+    } catch (UnauthorizedException e) {
+        return createAuthResultFromUnauthorizedExp(e);
+    } catch (Exception e) {
+      throw AuthEnforceUtil.propagateAccessException(e);
+    }
+    return AuthorizationResponse.Builder.defaultNotRequired();
+  }
+
+  @Override
+  public AuthorizationResponse dropRole(Principal caller, Role role) throws AccessException {
+    try {
+      accessControllerV1.dropRole(role);
+    } catch (UnauthorizedException e) {
+      return createAuthResultFromUnauthorizedExp(e);
+    } catch (Exception e) {
+      throw AuthEnforceUtil.propagateAccessException(e);
+    }
+    return AuthorizationResponse.Builder.defaultNotRequired();
+  }
+
+  @Override
+  public AuthorizationResponse addRoleToPrincipal(Principal caller, Role role, Principal principal)
+    throws AccessException {
+    try {
+      accessControllerV1.addRoleToPrincipal(role, principal);
+    } catch (UnauthorizedException e) {
+      return createAuthResultFromUnauthorizedExp(e);
+    } catch (Exception e) {
+      throw AuthEnforceUtil.propagateAccessException(e);
+    }
+    return AuthorizationResponse.Builder.defaultNotRequired();
+  }
+
+  @Override
+  public AuthorizationResponse removeRoleFromPrincipal(Principal caller, Role role, Principal principal)
+    throws AccessException {
+    try {
+      accessControllerV1.removeRoleFromPrincipal(role, principal);
+    } catch (UnauthorizedException e) {
+      return createAuthResultFromUnauthorizedExp(e);
+    } catch (Exception e) {
+      throw AuthEnforceUtil.propagateAccessException(e);
+    }
+    return AuthorizationResponse.Builder.defaultNotRequired();
+  }
+
+  @Override
+  public AuthorizationResult<Set<Role>> listRoles(Principal caller, Principal principal) throws AccessException {
+    Set<Role> roleSet;
+    try {
+      roleSet = accessControllerV1.listRoles(principal);
+    } catch (UnauthorizedException e) {
+      //Incase of UnauthorizedException , the set will be empty
+      return new AuthorizationResult<>(Collections.emptySet(), AuthorizationResponse.Builder.defaultNotRequired());
```

Comment:
I think you need to build a proper AuthrizationResponse. E.g. `return new AuthorizationResult<>(Collections.emptySet(),  createAuthResultFromUnauthorizedExp(e))

Reply:
Same in other places